python-lib/dku_data_processing/format.py

```diff
@@ -31,6 +32,62 @@ def wkt_parser(wkt_point):
         return result[0]
 
 
+def extract_df(df, detail, filters, geopoint, tooltips):
+    """
+
+    Extract and select only necessary data to send to front-end.
+    Handle the filtering, normalizing, parsing and the tooltip.
+
+    :param detail:
+    :param df:
+    :param filters:
+    :param geopoint:
+    :param tooltips:
+    """
+    # Handle filtering
+    if (filters is not None) and (filters != []):
+        df = filter_dataframe(df, filters)
+    if df.empty:
+        return []
+
+    # Handle normalizing of detail column
+    if (detail is None) or (df.dtypes[detail] not in ['int64', 'float64']):
+        df['detail_'] = 1
+    else:
+        df['detail_'] = normalize(df[detail].values)
+    if (detail is not None) and (df.dtypes[detail] not in ['int64', 'float64']):
+        raise ValueError("The detail column should be a int or float type")
+
+    def convert(x):
+        if x is not None:
+            return float(x)
+        else:
+            return None
+
+    # Parse longitude and latitude from WKT geopoint column
+    parsed_lat_long_df = df[geopoint].apply(wkt_parser)
+    df['longitude_'] = parsed_lat_long_df.apply(lambda x: x[0]).apply(lambda x: convert(x))
```

Comment:
Be careful with collisions : what will you do if there's already a "longitude_" column?